diff --git a/common/desktop-exports b/common/desktop-exports
index 1319361..cb9cd5d 100644
--- a/common/desktop-exports
+++ b/common/desktop-exports
@@ -94,6 +94,7 @@ append_dir LD_LIBRARY_PATH $RUNTIME/usr/lib/$ARCH/libunity
 
 # Pulseaudio export
 append_dir LD_LIBRARY_PATH $RUNTIME/usr/lib/$ARCH/pulseaudio
+append_dir LD_LIBRARY_PATH $SNAP/usr/lib/alsa-lib
 
 # EGL vendor files on glvnd enabled systems
 [ -d /var/lib/snapd/lib/glvnd/egl_vendor.d ] && \
@@ -424,3 +425,6 @@ IBUS_CONFIG_PATH=$XDG_CONFIG_HOME/ibus
 mkdir -p $IBUS_CONFIG_PATH
 [ -d $IBUS_CONFIG_PATH/bus ] && rm -rf $IBUS_CONFIG_PATH/bus
 ln -sfn $REALHOME/.config/ibus/bus $IBUS_CONFIG_PATH
+
+prepend_dir LD_LIBRARY_PATH $SNAP/usr/lib/$ARCH/blas
+prepend_dir LD_LIBRARY_PATH $SNAP/usr/lib/$ARCH/lapack
diff --git a/gtk/launcher-specific b/gtk/launcher-specific
index 3d2a8ff..82d5555 100644
--- a/gtk/launcher-specific
+++ b/gtk/launcher-specific
@@ -12,10 +12,8 @@ if [ "$USE_gtk3" = true ]; then
     # Does not hurt to specify this as well, just in case
     export QT_QPA_PLATFORM=wayland-egl
   fi
-  append_dir GTK_PATH $RUNTIME/usr/lib/$ARCH/gtk-3.0
 else
   [ "$wayland_available" = true ] && echo "Warning: GTK2 does not support Wayland!"
-  append_dir GTK_PATH $RUNTIME/usr/lib/$ARCH/gtk-2.0
 fi
 
 # ibus and fcitx integration
